exploration-module/src/main/java/io/edpn/backend/exploration/adapter/persistence/entity/mapper/MybatisSystemEntityMapper.java

```diff
@@ -0,0 +1,42 @@
+package io.edpn.backend.exploration.adapter.persistence.entity.mapper;
+
+import io.edpn.backend.exploration.adapter.persistence.entity.MybatisSystemEntity;
+import io.edpn.backend.exploration.application.domain.Coordinate;
+import io.edpn.backend.exploration.application.domain.System;
+import io.edpn.backend.exploration.application.dto.SystemEntity;
+import io.edpn.backend.exploration.application.dto.mapper.SystemEntityMapper;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+
+import java.util.Optional;
+
+@RequiredArgsConstructor
+@Slf4j
+public class MybatisSystemEntityMapper implements SystemEntityMapper<MybatisSystemEntity> {
+
+    @Override
+    public System map(SystemEntity systemEntity) {
+        return new System(
+                systemEntity.getId(),
+                systemEntity.getEliteId(),
+                systemEntity.getName(),
+                systemEntity.getStarClass(),
+                coordinateFromCoordinateEntity(systemEntity));
+    }
+
+    @Override
+    public MybatisSystemEntity map(System system) {
+        return new MybatisSystemEntity(
+                system.id(),
+                system.name(),
+                system.eliteId(),
+                system.starClass(),
+                Optional.ofNullable(system.coordinate()).map(Coordinate::x).orElse(null),
+                Optional.ofNullable(system.coordinate()).map(Coordinate::y).orElse(null),
+                Optional.ofNullable(system.coordinate()).map(Coordinate::z).orElse(null));
```

Comment:
In what situation would a system not have an x/y/z co-ordinate?

Should these not be non-null? Maybe we know the system name, but not it's location (thanks to EDDN design decisions)? I know there are some message types that state the system name but not it's location (my choice to address that in the past, was to make System have an `address` property, which may or may not be null... but an address must always have x/y/z).